[WIP] New minimal

numberZero opened the pull request:
WIP to fix #5192.

**MT weirdness found so far:**
* Solid nodes with transparent texture look solid in world (as they should), but transparent in inventory.
* Liquids with `use_texture_alpha` and low texture alpha are not visible in inventory. (#5461)
* Glasslike_framed is just weird when combined with facedir.
* Torchlike requires inventory and wield images.
* Signlike requires inventory image.
* Connected nodeboxes connect wrongly.
* Translucent overlay can’t be used on non-translucent base texture. (#6726)
* Plantlike/leveled textures are not tileable_vertical by default.
* Plantlike/leveled doesn’t support leveled selection and collision boxes.

sfan5 commented:
>removing `experimental`

:angry:

nerzhul commented:
don't wipe experimental it has some integration tests for CSM and SSM APIs

numberZero commented:
> don't wipe experimental it has some integration tests for CSM and SSM APIs

This is WIP. But thank you for pointing that, I could miss some tests. Surely I will restore them; the very essence of MDT is MT testing.

Ferk commented:
Other than one example for each drawtype, something that might be nice to have in minimal is maybe some ice/glass block with alpha transparency (maybe just reuse the water texture) that can be used to showcase and test the alpha sorting problems in the engine. Both MTG and minimal only have alpha in liquids, sometimes it's good to have a solid block to look through and check it.

paramat commented:
The mapgen aliases necessary are described in lua_api.txt now https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L238

numberZero commented:
Is it normal for minimal to expose some existing weird behavior?
![screenshot_20171201_214751](https://user-images.githubusercontent.com/7352626/33911016-dca2b6dc-dfa1-11e7-8dc3-1c735a558b07.png)
^ Inability to use translucent overlay on non-translucent node.
![screenshot_20171209_205124](https://user-images.githubusercontent.com/7352626/33911017-dccd050e-dfa1-11e7-98b3-1f5b65d4a25f.png)
^ Some mapgen problem (maybe missing aliases; if so, will not present)
![screenshot_20171210_232215](https://user-images.githubusercontent.com/7352626/33911018-dcf514b8-dfa1-11e7-8d0a-f021864b03f3.png)
^ Not obeying connection rules *sometimes*.

rubenwardy commented:
Make sure you have all the mapgen aliases, even if defined to air or stone, whichever is appropriate

https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L238

Wuzzy2 commented:
How is it going with this?
Also, do you have a TO-DO list? And what are the goals?

I'm not sure where you're going with this. I think any replacement for minimal needs to be useful to developers somehow. The core devs might have different priorities.

numberZero commented:
> How is it going with this?

Stalled. But even in its current state it’s useful for testing (like “does it work at all?”). And it exposed some tricky MT behavior.

> what are the goals

> I think MDT could be useful to build simple test cases for all the core features. E.g. for node drawtypes.
And if one feature breaks, all you have to do is to fire up MDT and look up the feature that is supposedly broken.

Mostly that.

Now I think I chosen a wrong way, developing `minimal` like a subgame. It is not a subgame, but a test suite instead. I will probably rewrite it that way (basically that means no crafting, but that certainly needs a sort of creative inventory, maybe even accessible to everyone).

So, thank you for reminding; I almost forgot that I started this.

numberZero commented:
See first post for a list of weirdness found during developing of this.

MarkuBu commented:
I don't know what you mean by "minimal". I stripped MTG to something real minimal

https://github.com/MarkuBu/minetest_rolad

And even this is not absolutely minimal because I added the flowlib mod and I didn't remove some files in default.

numberZero commented:
`minimal` is “Minimal Development Test.” But that’s a bit confusing so currently my stuff is named `devtest`, and has no gameplay (e.g. no crafting), only example nodes. Also currently it shows some edge cases where MT behaves strangely. E.g. this glasslike_framed node is just rotated differently:
![screenshot_20180129_200143](https://user-images.githubusercontent.com/7352626/35523335-ac08c8dc-052f-11e8-9026-e02e31e4eae8.png)
but uses different textures for the frame for some reason. I do understand that that drawtype was not intended to be rotatable, but it is not documented as not supporting that.

numberZero commented:
Removed overlays translucency as unsupported.
Added signlike (colored!) and plantlike (with meshoptions ofc) examples:
![screenshot_20180202_022003](https://user-images.githubusercontent.com/7352626/35708627-0794e8e2-07c0-11e8-9686-ed88197d37fa.png)
![screenshot_20180202_021309](https://user-images.githubusercontent.com/7352626/35708540-b0781e62-07bf-11e8-93e2-e04d5bacec09.png)

red-001 commented:
I suggest minimal is removed once devtest is finished, no reason to keep it at that point.

red-001 commented on games/devtest/mods/land/liquid.lua:
```diff
@@ -0,0 +1,54 @@
+local function register_liquid(name, description, texture, props, sprops)
```
might be nice to add a similar function to core.

paramat commented:
This will be very useful, i will certainly support this.

Wuzzy2 commented:
Do you have a TODO list for when you consider your work finished?

numberZero commented:
> Do you have a TODO list for when you consider your work finished?

No, except of drawtypes.

numberZero commented:
Note: this `devtest` is extremely useful for me already, despite lacking some stuff like creative inventory (very handy specifically for testing). So I decided to publish it as a separate repo: [devtest](https://github.com/numberZero/minetest-devtest) for now.
One question: what should I keep from the existing `minimal`? `experimental` (mostly), anything else?

numberZero commented:
Can’t finish in any foreseeable future. Current version is available at https://github.com/numberZero/minetest-devtest.
